q2_metadata/plugin_setup.py

```diff
@@ -59,3 +64,86 @@
                 'visualization supports interactive filtering, sorting, and '
                 'exporting to common file formats.',
 )
+
+ArtificialGrouping = \
+    SemanticType('ArtificialGrouping', variant_of=SampleData.field['type'])
+
+plugin.register_semantic_types(ArtificialGrouping)
+
+
+class ArtificialGroupingFormat(model.TextFileFormat):
+    def _validate(self, n_records=None):
+        with self.open() as fh:
+            # validate header
+            line = fh.readline()
+            line.startswith('sample-id')
```

Comment:
`sample-id` isn't guaranteed since you're inheriting whatever the original index's name was.

It might be best to use q2.Metadata directly and save a lot of trouble. Here's a really simple version you can adapt:
https://github.com/qiime2/q2-types/blob/master/q2_types/feature_data/_format.py#L339-L351

Reply:
Makes sense, thanks for the link!